backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java

```diff
@@ -228,7 +231,11 @@ private static void copyDiscountIntoResultIfAllowedByPricingContext(
 	{
 		if (!pricingCtx.isDisallowDiscount())
 		{
-			result.setDiscount(subscriptionPricingResult.getDiscount());
+			if (result.isDiscountEditable())
+			{
+				result.setDiscount(subscriptionPricingResult.getDiscount());
+				result.setDiscountEditable(subscriptionPricingResult.isDiscountEditable());
```

Comment:
i would say this `result.setDiscountEditable` still makes sense..and as a 3rd line, here we should also copy the `dontOverrideDiscountAdvice` or whatever property you end up with.

Reply:
added.